app/scripts/directives/updateTimestamp.js

```diff
@@ -0,0 +1,224 @@
+'use strict';
+
+/**
+ * @ngdoc directive
+ * @name oncokb.directive:updateTimestamp
+ * @author Jing Su created on 2019/06/26
+ * @description
+ * # refresh date
+ */
+angular.module('oncokbApp')
+    .directive('updateTimestamp', function(_, $timeout, mainUtils, DatabaseConnector, FirebaseModel, $rootScope, firebaseConnector) {
+        return {
+            templateUrl: 'views/updateTimestamp.html',
+            restrict: 'E',
+            scope: {
+                type: '=',
+                hugoSymbol: '=',
+                key: '=',
+                obj: '=',
+                disabled: '=',
+                updateTime: '=',
+                updatedBy: '=',
+                path: '='
+            },
+            replace: false,
+            controller: function($scope) {
+                $scope.clicked = false;
+                $scope.getIconClass = function(time) {
+                    if ($scope.disabled) return 'disabled';
+                    else return mainUtils.getTimestampClass(time);
+                };
+                $scope.validateTime = function () {
+                    $scope.clicked = true;
+                    $scope.errorMessage = '';
+                    if ($scope.type === 'tools') {
+                        validateTimeInTools();
+                    } else {
+                        mainUtils.validateTime($scope.obj, [$scope.key]);
+                        if ($scope.key === 'type') {
+                            firebaseConnector.set($scope.path + '/' + $scope.key + '_validateTime', $scope.obj[$scope.key + '_validateTime']).then(function (result) {
+                                $scope.updateTime = $scope.obj[$scope.key + '_validateTime'].updateTime;
+                                $scope.clicked = false;
+                            }, function (error) {
+                                $scope.clicked = false;
+                                console.log("Error:", error);
+                            });
+                        } else {
+                            var data = {};
+                            data[$scope.obj[$scope.key + '_uuid']] = $scope.obj[$scope.key + '_validateTime'].updateTime;
+                            DatabaseConnector.updateEvidenceLastReview(data, function(){
+                                $scope.updateTime = $scope.obj[$scope.key + '_validateTime'].updateTime;
+                                $scope.clicked = false;
+                                firebaseConnector.set($scope.path + '/' + $scope.key + '_validateTime', $scope.obj[$scope.key + '_validateTime']).then(function (result) {
+                                }, function (error) {
+                                    console.log("Error:", error);
```

Comment:
As I mentioned earlier. These console logs will not be seen. You should properly handle the reaction, giving a window popup for instance, if the error is critical.

Reply:
Added popup window when update timestamp service failed.